File: CS/GrowthWare.WebApplication/UserControls/FileManagerControl.ascx.cs
using GrowthWare.Framework.Model.Profiles;
using GrowthWare.WebSupport.Utilities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;

namespace GrowthWare.WebApplication.UserControls
{
    public partial class FileManagerControl : System.Web.UI.UserControl
    {
        protected void Page_Load(object sender, EventArgs e)
        {
            SearchControl.ShowAddLink = false;
            MDirectoryProfile mDirectoryProfile = DirectoryUtility.GetProfile(FunctionUtility.CurrentProfile().Id);
            String mLinks = GrowthWare.WebApplication.Functions.System.FileManagement.FileManager.GetDirectoryLinks("/", mDirectoryProfile.FunctionSeqId);
            directorySelector.InnerHtml = mLinks;
            MFunctionProfile mFunctionProfile = FunctionUtility.CurrentProfile();
            MAccountProfile mAccountProfile = AccountUtility.CurrentProfile();
            MSecurityInfo mSI = new MSecurityInfo(mFunctionProfile, mAccountProfile);
            UploadControl.Visible = mSI.MayAdd;
            tdNewDirectory.Visible = mSI.MayAdd;
            SearchControl.ShowDeleteAll = mSI.MayDelete;
            SearchControl.ShowSelect = mSI.MayDelete;
        }
    }
}

File: VB/GrowthWare.WebApplication/UserControls/FileManagerControl.ascx.vb
Imports GrowthWare.Framework.Model.Profiles
Imports GrowthWare.WebSupport.Utilities

Public Class FileManagerControl
    Inherits System.Web.UI.UserControl

    Protected Sub Page_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        SearchControl.ShowAddLink = False
        Dim mDirectoryProfile As MDirectoryProfile = DirectoryUtility.GetProfile(FunctionUtility.CurrentProfile().Id)
        Dim mLinks As String = FileUtility.GetDirectoryLinks("/", mDirectoryProfile.FunctionSeqId)
        directorySelector.InnerHtml = mLinks
        Dim mFunctionProfile As MFunctionProfile = FunctionUtility.CurrentProfile()
        Dim mAccountProfile As MAccountProfile = AccountUtility.CurrentProfile()
        Dim mSI As MSecurityInfo = New MSecurityInfo(mFunctionProfile, mAccountProfile)
        UploadControl.Visible = mSI.MayAdd
        tdNewDirectory.Visible = mSI.MayAdd
        SearchControl.ShowDeleteAll = mSI.MayDelete
        SearchControl.ShowSelect = mSI.MayDelete
    End Sub

End Class